cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClientService.java

```diff
@@ -40,8 +41,12 @@ public AzureClient getClient(CloudCredential cloudCredential) {
 
     public AzureClient getClient(CloudContext cloudContext, CloudCredential cloudCredential) {
         AzureCredentialView azureCredentialView = new AzureCredentialView(cloudCredential);
-        AzureClientCredentials azureClientCredentials = new AzureClientCredentials(cloudContext, azureCredentialView, logLevel, cbRefreshTokenClientProvider,
-                authenticationContextProvider);
-        return new AzureClient(azureClientCredentials, azureAuthExceptionHandler);
+        if (AppAuthenticationType.CERTIFICATE.name().equals(azureCredentialView.getAuthenticationType())
+                && AppCertificateStatus.KEY_GENERATED.name().equals(azureCredentialView.getStatus())) {
+            return null;
```

Comment:
is it possible to handle this with Optional? returning with null looks weird and error prone

Reply:
I would move this part to the `createAuthenticatedContext` method but not make it optional. It requires lot of changes to make it optional and the original problem is that we shouldn't authenticate when the credential is in KEY_GENERATED status. Unfortunately it would require adding other methods to cloud-api module.

Reply:
ok